src/item_pocket.h

```diff
@@ -0,0 +1,174 @@
+#pragma once
+#ifndef ITEM_POCKET_H
+#define ITEM_POCKET_H
+
+#include <list>
+
+#include "enums.h"
+#include "enum_traits.h"
+#include "optional.h"
+#include "type_id.h"
+#include "ret_val.h"
+#include "translations.h"
+#include "units.h"
+#include "visitable.h"
+
+class Character;
+class item;
+class item_location;
+class player;
+
+struct iteminfo;
+struct itype;
+struct tripoint;
+
+using itype_id = std::string;
+
+class item_pocket
+{
+    public:
+        enum pocket_type {
+            // this is to aid the transition from the previous way item contents were handled.
+            // this will have the rules that previous contents would have
+            LEGACY_CONTAINER,
+            CONTAINER,
+            MAGAZINE,
+            LAST
+        };
+        enum contain_code {
```

Comment:
You're already treating these `enum`s as if they were `enum class` (by always including the `contain_code::` before each enumerator name, so perhaps declare them thus?